### PR TITLE
feat(metrics): service name change and switching the metrics default to otelcol

### PR DESCRIPTION
**Pull request by rnishtala-sumo**
##### Description

- Service name change and switching the metrics default to otelcol
---

##### Checklist

- [x] Changelog updated

###### Testing performed

- [x] Redeploy otelcol-logs and otelcol-metrics pods
- [ ] Confirm logs and metrics are coming in

### CODE REVIEW

File: tests/integration/helm_traces_enabled_test.go [hunk 1]
```diff
@@ -156,16 +156,6 @@ func Test_Helm_Traces_Enabled(t *testing.T) {
...
-		Assess("prometheus pods are available",
-			stepfuncs.WaitUntilPodsAvailable(
-				metav1.ListOptions{
-					LabelSelector: "app.kubernetes.io/name=prometheus",
-				},
-				1,
-				waitDuration,
-				tickDuration,
-			),
-		).
```
[the remark this answers is not in the document]
Comment (rnishtala-sumo) [reply tests/integration/helm_traces_enabled_test.go:168]: For some reason, after switching to otel, this test fails with:

```
Warning   FailedScheduling        pod/prometheus-rel-3497023499-kube-promet-prometheus-0                          0/1 nodes are available: 1 Insufficient cpu.
```

File: deploy/helm/sumologic/templates/_helpers.tpl [hunk 2]
```diff
@@ -460,11 +460,11 @@ helm.sh/hook-delete-policy: before-hook-creation,hook-succeeded
 {{- end -}}
 
 {{- define "sumologic.metadata.name.metrics.service" -}}
-{{ template "sumologic.metadata.name.fluentd" . }}-metrics
+{{ template "sumologic.fullname" . }}-metadata-metrics
 {{- end -}}
 
 {{- define "sumologic.metadata.name.metrics.service-headless" -}}
-{{ template "sumologic.metadata.name.metrics.service" . }}-headless
+{{ template "sumologic.metadata.name.fluentd" . }}-metrics-headless
```
[the remark this answers is not in the document]
Comment (rnishtala-sumo) [reply deploy/helm/sumologic/templates/_helpers.tpl:467]: My understanding is that in v2, the headless service name is always called `fluentd`, regardless of the metadata provider. Is that not correct?

[the remark this answers is not in the document]
Comment (swiatekm-sumo) [reply deploy/helm/sumologic/templates/_helpers.tpl:467]: I don't think so, the Pods are actually named `*-otelcol`.

[the remark this answers is not in the document]
Comment (rnishtala-sumo) [reply deploy/helm/sumologic/templates/_helpers.tpl:467]: yes the pods are *-otelcol, but the service name still fluentd-headless. That's what I observed in this PR in v2
https://github.com/SumoLogic/sumologic-kubernetes-collection/pull/1755

File: tests/integration/helm_traces_enabled_test.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (rnishtala-sumo) [reply tests/integration/helm_traces_enabled_test.go:168]: I'm keeping some of the installation tests, to ensure that the otagent is up and running before we attempt to send traces.

**Review by swiatekm-sumo — APPROVED** [review]

**Pull request merged**